rtrlib/rtr_mgr.c

```diff
@@ -748,3 +751,58 @@ RTRLIB_EXPORT inline void rtr_mgr_for_each_ipv6_record(
 	pfx_table_for_each_ipv6_record(config->pfx_table,
 				       fp, data);
 }
+
+#ifdef RTRLIB_BGPSEC
+RTRLIB_EXPORT int rtr_mgr_bgpsec_validate_as_path(
+				const struct rtr_bgpsec_data *data,
+				const struct rtr_signature_seg *sig_segs,
+				const struct rtr_secure_path_seg *sec_paths,
+				struct rtr_mgr_config *config,
+				const unsigned int as_hops)
+{
+	int retval = rtr_bgpsec_validate_as_path(data,
+						 sig_segs,
+						 sec_paths,
+						 &config->spki_table,
+						 as_hops);
+
+	return retval;
+}
+
+RTRLIB_EXPORT int rtr_mgr_bgpsec_generate_signature(
+				  const struct rtr_bgpsec_data *data,
+				  const struct rtr_signature_seg *sig_segs,
+				  const struct rtr_secure_path_seg *sec_paths,
+				  const unsigned int as_hops,
+				  const struct rtr_secure_path_seg *own_sec_path,
+				  const unsigned int target_as,
+				  uint8_t *private_key,
+				  uint8_t *new_signature)
+{
+	int retval = rtr_bgpsec_generate_signature(data,
+						   sig_segs,
+						   sec_paths,
+						   as_hops,
+						   own_sec_path,
+						   target_as,
+						   private_key,
+						   new_signature);
+
+	return retval;
+}
+
+RTRLIB_EXPORT int rtr_mgr_bgpsec_get_version(void)
+{
+	return rtr_bgpsec_get_version();
+}
+
+RTRLIB_EXPORT int rtr_mgr_bgpsec_has_algorithm_suite(unsigned int alg_suite)
```

Comment:
why `unsigned int` and `uint8_t` for `get_algoritm_suites`

Reply:
No particular reason. Suggestions?

Reply:
Since algorithm suites are specified to be 1 byte long, I will change it to `uint8_t`.